### PR TITLE
Add support for`helm list --no-headers`

### LINKED ISSUE #10548: Add support for `helm list --no-headers`
Add support for helm list --no-headers.

One frequently used kubectl option is --no-headers, it would be more convenient for users if helm also supports it.

### PR DESCRIPTION
**Pull request by suzaku**
**What this PR does / why we need it**:

Fix #10548 

Some other changes in the PR:

1. `newReleaseListWriter` is changed to return a struct instead, because it's a tiny struct and we don't need to update it in place, so there's no need to use pointer semantic here.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by yxxhero** [comment]
@suzaku Please create a issue for this PR. Thanks very much.

**Comment by yxxhero** [comment]
@suzaku unittest is required.

**Comment by suzaku** [comment]
> @suzaku unittest is required.

Test added.

**Comment by suzaku** [comment]
@yxxhero Ping.

**Comment by suzaku** [comment]
@yxxhero I've added the tests.

**Comment by yxxhero** [comment]
@suzaku You should make minimal changes to add this feature, so you should keep the original pointer return. What do you think of.

**Comment by suzaku** [comment]
> @suzaku You should make minimal changes to add this feature, so you should keep the original pointer return. What do you think of.

I can do that in a separate PR.

**Comment by yxxhero** [comment]
@suzaku  Why not in this PR? Thanks very much.

**Comment by suzaku** [comment]
> @suzaku Why not in this PR? Thanks very much.

I mean I can change the return value in a separate PR.

**Comment by suzaku** [comment]
@yxxhero Updated.

**Comment by yxxhero** [comment]
@suzaku Thanks very much. all is ok. Please sqush your commit history.

**Comment by suzaku** [comment]
@yxxhero Done.

**Comment by yxxhero** [comment]
@bacongobbler @scottrigby ping

**Comment by suzaku** [comment]
@yxxhero Don't you have the permission to approve this PR?

**Comment by yxxhero** [comment]
@suzaku wait for core maintainers. Thanks very much.

**Review by joejulian — APPROVED** [review]
This looks good to me.

This will still need approval from a core maintainer, of which I am not.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @suzaku for this.

**Comment by hickeyma** [comment]
Needs one more helm maintainer to merge.

**Review by scottrigby — APPROVED** [review]
lgtm!

**Comment by suzaku** [comment]
Thanks for reviewing this PR.
Could you please also take a look at this PR: #10564

**Pull request merged**
